apps/src/templates/instructions/TopInstructionsCSF.jsx

```diff
@@ -674,16 +674,21 @@ var TopInstructions = React.createClass({
                 </LegacyButton>
               }
             </ChatBubble>
-            {!this.props.collapsed && this.props.hints && this.props.hints.map((hint) =>
-              <InlineHint
-                key={hint.hintId}
-                borderColor={color.yellow}
-                content={hint.content}
-                ttsUrl={hint.ttsUrl}
-                ttsMessage={hint.ttsMessage}
-                block={hint.block}
-              />
-            )}
+            {!this.props.collapsed && this.props.hints && this.props.hints.map((hint) => {
+              if (hint.hintPath) {
+                window.dispatchEvent(new CustomEvent('displayHintPath', {detail: hint.hintPath}));
```

Comment:
I'm really not a fan of a map operation having a side effect. Can the craft app not subscribe to the redux store directly?

Reply:
The problem is that Redux really wants you to subscribe via `connect`.  See: https://github.com/reactjs/redux/issues/580#issuecomment-133188511.

I can do `store.subscribe()` to get notified on *every single* Redux action, then look at `store.getState().authoredHints.seenHints` to see if a hint is visible that doesn't match a cached list.  Do you prefer that approach?